contracts/NonFungibleToken-v2.cdc

```diff
@@ -0,0 +1,257 @@
+It includes a few functions to allow the owner to easily
+move tokens in and out of the collection.
+
+## `Provider` and `Receiver` resource interfaces
+
+These interfaces declare functions with some pre and post conditions
+that require the Collection to follow certain naming and behavior standards.
+
+They are separate because it gives the user the ability to share a reference
+to their Collection that only exposes the fields and functions in one or more
+of the interfaces. It also gives users the ability to make custom resources
+that implement these interfaces to do various things with the tokens.
+
+By using resources and interfaces, users of NFT smart contracts can send
+and receive tokens peer-to-peer, without having to interact with a central ledger
+smart contract.
+
+To send an NFT to another user, a user would simply withdraw the NFT
+from their Collection, then call the deposit function on another user's
+Collection to complete the transfer.
+
+*/
+
+import MetadataViews from "./MetadataViews.cdc"
+
+/// The main NFT contract interface. Other NFT contracts will
+/// import and implement this interface
+///
+pub contract interface NonFungibleToken {
+
+    /// Event that is emitted when a token is withdrawn,
+    /// indicating the owner of the collection that it was withdrawn from.
+    ///
+    /// If the collection is not in an account's storage, `from` will be `nil`.
+    ///
+    pub event Withdraw(id: UInt64, from: Address?, type: Type, 
+                       displayView: MetadataViews.Display?, 
+                       serialView: MetadataViews.Serial?)
+
+    /// Event that emitted when a token is deposited to a collection.
+    ///
+    /// It indicates the owner of the collection that it was deposited to.
+    ///
+    pub event Deposit(id: UInt64, to: Address?, type: Type,
+                      displayView: MetadataViews.Display?,
+                      serialView: MetadataViews.Serial?)
+
+    /// Transfer
+    ///
+    /// The event that should be emitted when tokens are transferred from one account to another
+    ///
+    pub event Transfer(id: UInt64, from: Address?, to: Address?, type: Type,
+                       displayView: MetadataViews.Display?,
+                       serialView: MetadataViews.Serial?)
+
+    /// Mint
+    ///
+    /// The event that should be emitted when an NFT is minted
+    pub event Mint(id: UInt64, type: Type,
+                   displayView: MetadataViews.Display?,
+                   serialView: MetadataViews.Serial?)
+
+    /// Destroy
+    ///
+    /// The event that should be emitted when an NFT is destroyed
+    pub event Destroy(id: UInt64, type: Type,
+                      displayView: MetadataViews.Display?,
+                      serialView: MetadataViews.Serial?)
+
+    /// Interface that the NFTs must conform to
+    ///
+    pub resource interface NFT: MetadataViews.Resolver {
+        /// The unique ID that each NFT has
+        pub fun getID(): UInt64
```

Comment:
I think this shoukd return uuid if that is possible.

Reply:
added